### PR TITLE
Add an option to change stream push notifications in StreamScreen

### PR DESCRIPTION
**Pull request by kunall17**
simulator screen shot - iphone 6 - 2017-12-13 at 23 08 57

### CODE REVIEW

File: src/common/OptionRow.js [hunk 1]
```diff
@@ -17,17 +18,18 @@ const styles = StyleSheet.create({
 type Props = {
   label: string,
   defaultValue: boolean,
+  customStyle?: StyleObj,
```
Comment (borisyankov) [inline src/common/OptionRow.js]: Just call this `style`.

File: src/streams/StreamScreen.js [hunk 2]
```diff
@@ -22,22 +25,40 @@ class StreamScreen extends PureComponent<Props> {
...
     const stream = streams.find(x => x.stream_id === streamId);
-    const subscription = subscriptions.find(x => x.stream_id === streamId);
+    const subscription = subscriptions.find(x => x.stream_id === streamId) || NULL_SUBSCRIPTION;
 
     return (
       <Screen title="Stream" padding>
         <StreamCard stream={stream} subscription={subscription} />
         <ZulipButton text="Edit" onPress={this.handleEdit} />
+        <OptionRow
+          label="Push notifications"
```
Comment (borisyankov) [inline src/streams/StreamScreen.js]: Just use `Notifications`, I am intentionally using the shorter, simpler term (and no need to translate it then)

File: src/streams/StreamScreen.js [hunk 3]
```diff
@@ -22,22 +25,40 @@ class StreamScreen extends PureComponent<Props> {
...
 
+  static contextTypes = {
+    styles: () => null,
+  };
+
+  toggleStreamPushNotification = () => {
+    const { subscriptions, navigation, auth } = this.props;
+    const { streamId } = navigation.state.params;
+    const subscription = subscriptions.find(x => x.stream_id === streamId) || NULL_SUBSCRIPTION;
+    toggleStreamNotification(auth, streamId, !subscription.push_notifications);
```
Comment (borisyankov) [inline src/streams/StreamScreen.js]: Add an `s` to the name as `toggleStreamNotifications`?

Also, extract that to an action and retrieve the `auth` and `subscription` from the store there.

**Comment by kunall17** [comment]
Updated!

**Pull request merged**